src/api/test-controller/index.js

```diff
@@ -379,6 +379,13 @@ export default class TestController {
     _expect$ (actual) {
         const callsite = getCallsiteForMethod('expect');
 
+        if (global.snapshotPropertyCallsite && global.snapshotPropertyCallsite.filename === callsite.filename &&
```

Comment:
It's better to use `Symbol` when creating global properties. As a better alternative, you can use a module-scoped variable for this task. Let's extract this callsite tracker and `callsitesWithoutAwait` to a module.

Reply:
I like this approach. I do not have other suggestings. I'll check this PR one more time after modifications.